nucleus/modelci/eval_functions/available_eval_functions.py

```diff
@@ -0,0 +1,189 @@
+from typing import Callable, Dict, List, Type, Union
+
+from nucleus.logger import logger
+from nucleus.modelci.eval_functions.base_eval_function import BaseEvalFunction
+
+from ..data_transfer_objects.eval_function import EvalFunctionEntry
+from ..errors import EvalFunctionNotAvailableError
+
+MEAN_AVG_PRECISION_NAME = "mean_average_precision_boxes"
+
+
+class BoundingBoxIOU(BaseEvalFunction):
+    @classmethod
+    def expected_name(cls) -> str:
+        return "bbox_iou"
+
+
+class BoundingBoxMeanAveragePrecision(BaseEvalFunction):
+    @classmethod
+    def expected_name(cls) -> str:
+        return "bbox_map"
+
+
+class BoundingBoxRecall(BaseEvalFunction):
+    @classmethod
+    def expected_name(cls) -> str:
+        return "bbox_recall"
+
+
+class BoundingBoxPrecision(BaseEvalFunction):
+    @classmethod
+    def expected_name(cls) -> str:
+        return "bbox_precision"
+
+
+class CustomEvalFunction(BaseEvalFunction):
+    @classmethod
+    def expected_name(cls) -> str:
+        return "private_custom_function"  # Placeholder: See super().eval_func_entry for actual name
```

Comment:
Can this raise an error when not implemented (NotImplementedError?) bit confused about this